R/data.table.R

```diff
@@ -2849,7 +2842,7 @@ setDT = function(x, keep.rownames=FALSE, key=NULL, check.names=FALSE) {
   }
   # check no matrix-like columns, #3760. Other than a single list(matrix) is unambiguous and depended on by some revdeps, #3581
   if (length(x)>1L) {
-    idx = vapply_1i(x, function(xi) length(dim(xi)))>1L
+    idx = cols_with_dims(x)
```

Comment:
wild that this change is apparently causing some .internal.selfref issue. I've checked `idx` is the same with both approaches. I'm just going to revert this for now.